pkg/gauges/contracts/BalancerMinter.sol

```diff
@@ -0,0 +1,178 @@
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeMath.sol";
+
+import "./interfaces/IBalancerToken.sol";
+import "./interfaces/IGaugeController.sol";
+import "./interfaces/ILiquidityGauge.sol";
+
+contract BalancerMinter {
+    using SafeMath for uint256;
+
+    IBalancerToken private immutable _token;
+    IGaugeController private immutable _controller;
+
+    event Minted(address indexed recipient, address gauge, uint256 minted);
+
+    // user -> gauge -> value
+    mapping(address => mapping(address => uint256)) private _minted;
+    // minter -> user -> can mint?
+    mapping(address => mapping(address => bool)) private _allowedMinter;
+
+    constructor(IBalancerToken token, IGaugeController controller) {
+        _token = token;
+        _controller = controller;
+    }
+
+    /**
+     * @notice Mint everything which belongs to `msg.sender` and send to them
+     * @param gauge `LiquidityGauge` address to get mintable amount from
+     */
+    function mint(address gauge) external {
+        _mintFor(gauge, msg.sender);
+    }
+
+    /**
+     * @notice Mint everything which belongs to `msg.sender` across multiple gauges
+     * @param gauges List of `LiquidityGauge` addresses
+     */
+    function mintMany(address[] calldata gauges) external {
+        _mintForMany(gauges, msg.sender);
+    }
+
+    /**
+     * @notice Mint tokens for `user`
+     * @dev Only possible when `msg.sender` has been approved via `toggleApproveMint`
+     * @param gauge `LiquidityGauge` address to get mintable amount from
+     * @param user Address to mint to
+     */
+    function mintFor(address gauge, address user) external {
+        require(_allowedMinter[msg.sender][user], "Caller not allowed to mint for user");
+        _mintFor(gauge, msg.sender);
+    }
+
+    /**
+     * @notice Mint tokens for `user` across multiple gauges
+     * @dev Only possible when `msg.sender` has been approved via `toggleApproveMint`
+     * @param gauges List of `LiquidityGauge` addresses
+     * @param user Address to mint to
+     */
+    function mintManyFor(address[] calldata gauges, address user) external {
+        require(_allowedMinter[msg.sender][user], "Caller not allowed to mint for user");
+        _mintForMany(gauges, user);
+    }
+
+    /**
+     * @notice The total number of tokens minted for `user` from `gauge`
+     */
+    function minted(address user, address gauge) external view returns (uint256) {
+        return _minted[user][gauge];
+    }
+
+    /**
+     * @notice Whether `minter` is approved to mint tokens for `user`
+     */
+    function approvedMinter(address minter, address user) external view returns (bool) {
+        return _allowedMinter[minter][user];
+    }
+
+    /**
+     * @notice Set whether `minter` is approved to mint tokens on your behalf
+     */
+    function setMinterApproval(address minter, bool approval) public {
```

Comment:
Yeah, this seemed like a better interface than a simple toggle like in `toggle_approve_mint` to avoid accidentally allowing a minter when it's expected to disallow.